File: 71. Simplify Path/Java/Simplify Path.java
public class Solution {
    public String simplifyPath(String path) {
        List<String> stack = new ArrayList<>();
        for (String item : path.split("/")) {
            if (item.isEmpty() || item.equals(".")) {
                continue;
            }
            if (item.equals("..")) {
                if (!stack.isEmpty()) {
                    stack.remove(stack.size() - 1);
                }
            } else {
                stack.add(item);
            }
        }
        StringBuilder builder = new StringBuilder();
        for (String item : stack) {
            builder.append("/");
            builder.append(item);
        }
        if (builder.length() == 0) {
            builder.append("/");
        }
        return builder.toString();
    }
}

File: 71. Simplify Path/Python/Simplify Path.py
class Solution(object):
    def simplifyPath(self, path):
        """
        :type path: str
        :rtype: str
        """
        stack = []
        for item in path.split('/'):
            if item == '' or item == '.':
                continue
            if item == '..':
                if stack:
                    stack.pop()
            else:
                stack.append(item)
        result = '/'.join(stack)
        result = '/' + result
        return result
